enrolment-service/enrol/core/views.py

```diff
@@ -191,39 +187,62 @@ def retrieve(self, request, pk=None):
         return Response(status=status.HTTP_404_NOT_FOUND)
 
     def perform_update(self, serializer):
+        user = self.request.user
+        project = self.get_object()
+        previous_status = project.status
+        new_status = serializer.validated_data['status']
+        # project_before = serializer.instance
+
+        logger.debug('updating id %s', project.id)
+        logger.debug('updating previous status %s', project.status)
+        logger.debug('updating incoming %s', serializer.validated_data)
+        logger.debug('updating by user %s', user)
+
         # Save the data
-        project = serializer.save()
+        updated_project = serializer.save()
+        project_users = ProjectUsers.objects.filter(project=project.id)
+        project_users.update(status=new_status)
 
-        # If Termination Reqeust
-        if project.status == 3:
-            project_users = ProjectUsers.objects.filter(project=project.id)
-            project_users.update(status=3)
+        # Termination Request
+        if updated_project.status == 3:
             project_user_list = list(project_users.values())
 
             email = {
                 'to': RESOURCE_ADMIN_EMAIL,
                 'cc': project.user.email,
                 'subject': 'Project Termination Request from {} for project "{}"'.format(
-                    project.user.email, project.project_name),
+                    project.user.email,
+                    project.project_name
+                ),
                 'message': 'A request from email <strong>{}</strong> to terminate project <strong>"{}"</strong> has been initated. {}'.format(
-                    project.user.email, project.project_name, project_user_email_text(project_user_list))
+                    project.user.email,
+                    project.project_name,
+                    project_user_email_text(project_user_list)
+                )
             }
             send_update_notification(email)
-            # print(email) # for testing
 
-        elif project.status == 4:
-            project_users = ProjectUsers.objects.filter(project=project.id)
-            project_users.update(status=4)
-            email = {
-                'to': project.user.email,
-                'cc': RESOURCE_ADMIN_EMAIL,
-                'subject': 'Project Termination Request complete for project "{}"'.format(
-                    project.project_name),
-                'message': 'The request from email <strong>{}</strong> to terminate project <strong>"{}"</strong> is now complete.'.format(
-                    project.user.email, project.project_name)
-            }
-            send_update_notification(email)
-            # print(email) # for testing
+        # Termination Confirmation
+        elif updated_project.status == 4:
+
+            # confirming a request.
+            if previous_status == 3:
```

Comment:
Only use the existing "send a confirmation email" flow if the change is accepting a PI request.

Reply:
Useful code comments on these .status checks would be to declare what state each number is.

Reply:
Agreed! figured adding those throughout the file would be beyond scope, though.
These were originally defined in the model, and I'm just reusing the existing code.
I thought adding the intention declaratively in comments would suffice without scope creep:
e.g. "termination confirmation", "confirming a request"... but I guess I should make those a bit more verbose.

Let's see what I can do, without needing to go beyond the method definition.